common/camera.go

```diff
@@ -60,7 +62,18 @@ type CameraSystem struct {
 }
 
 // New initializes the CameraSystem.
-func (cam *CameraSystem) New(*ecs.World) {
+func (cam *CameraSystem) New(w *ecs.World) {
+	num := 0
+	for _, sys := range w.Systems() {
+		switch sys.(type) {
+		case *CameraSystem:
+			num++
+		}
+	}
+	if num > 0 { //initalizer is called before added to w.systems
+		panic("More than one CameraSystem was added to the World. The RenderSystem adds a CameraSystem if none exist when it's added.")
```

Comment:
Hi @Noofbiz, I dont think that panicking would be nice here (saw many panics in code) and I believe that It'd be better to just return an error so we can handle it later, or in the last case if you want to keep this behavior of `New`, a `log.Fatal(err)` would get rid of the stack trace and return only the error.

Reply:
@arxdsilva Upon reflection, I agree. An additional `CameraSystem` isn't going to wreck anything, just take up memory unnecessiarly. I think a `panic` or `log.Fatal(err)` would be going too far. A warning printed to the log without crashing the game should suffice.